src/nodes.coffee

```diff
@@ -270,7 +270,7 @@ exports.Base = class Base
   # as JSON. This is what the `ast` option in the Node API returns.
   # We try to follow the [Babel AST spec](https://github.com/babel/babel/blob/master/packages/babel-parser/ast/spec.md)
   # as closely as possible, for improved interoperability with other tools.
-  ast: ->
+  ast: (o) ->
```

Comment:
This PR introduces the `o` param, as it's needed as an argument for `.isStatement()`, which we need in order to be able to decide whether to wrap a block's expressions/statements in `ExpressionStatement` AST nodes or not

Reply:
Do you mind please explaining this a little more thoroughly? I've never fully understood the `o` param.

Reply:
Sure, I think you could call `o` a "context" param, basically an object that holds various configuration/other stuff relevant to the currently executing code (in this case, during AST generation inside `nodes.coffee`)

Whether doing AST generation (ie calling `nodes.ast(options)`) or transpile-to-JS (`nodes.compileToFragments(options)`), its initial value is the outer `options` argument to `compile()` in `coffeescript.coffee` (so basically the command-line options or equivalent per my understanding, eg `options.bare`)

Some of the big things that get passed around on `o` inside `nodes.coffee` are the current scope (`o.scope`) and the current compilation "level" (eg `LEVEL_LIST`). We haven't needed to pass around `o` yet while doing AST generation since we haven't introduced any logic yet that relies on things like scope or compilation level (but we will)

But now we need to use the existing `isStatement()` method to help determine the block-level AST (where non-statement "expressions" need to get wrapped in an `ExpressionStatement` AST node while statement nodes (eg `ExportNamedDeclaration`) don't)

Since `isStatement()` expects `o` as a param (to allow node classes to dynamically determine whether they're a statement or not based on things like current compilation level), we need to start passing `o` around now

So basically it gets passed down the same way as when transpiling to JS - the initial value of `o` gets passed in as the `options` from `compile()` in `coffeescript.coffee`, and then it just keeps getting passed down (and modified as appropriate) while we traverse the node tree (so primarily via calls to `.ast()`/`.compileNode()` etc)